### ISSUE TITLE
Breaks Google Closure type "optional" return type

### ISSUE THREAD

**Issue by turadg** [hosseinmd/prettier-plugin-jsdoc#153]
[TypeScript supports the Google Closure style in JSDoc](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#param-and-returns) to indicate a param is optional:
```js
/**
 * @param {string=} p2 - An optional param (Google Closure syntax)
 * @param {string} [p3] - Another optional param (JSDoc syntax).
 */
```

This formatter converts the former to the latter,
```js
/**
 * @param {string} [p2] - An optional param (Google Closure syntax) <-- NOT ANYMORE
 * @param {string} [p3] - Another optional param (JSDoc syntax).
 */
```

That's fine because it means the same thing. The problem is it also does this for return types:
```js
/** @returns {string=} */
```
becomes
```js
/** @returns {string} */
```

That changes the type of the return from `string | undefined` to `string`, a breaking change.

**Comment by hosseinmd** [comment]
We could fix it, but I think that could be more cool if you change it to `/** @returns {string | undefined} */`

**Comment by hosseinmd** [comment]
Maybe this parser should do it too.

**Comment by hosseinmd** [comment]
Fixed V0.3.32
